sdl_viewer/src/main.rs

```diff
@@ -211,21 +211,27 @@ impl NodeView {
 // Keeps track of the nodes that were requested in-order and loads then one by one on request.
 struct NodeViewContainer {
     node_views: HashMap<octree::NodeId, NodeView>,
-    requested: HashSet<octree::NodeId>,
+    requested: HashSet<octree::NodeId>,     // The node_ids that the I/O thread is currently loading.
+    // Communication with the I/O thread.
     node_id_sender: Sender<octree::NodeId>,
     node_data_receiver: Receiver<(octree::NodeId, octree::NodeData)>,
 }
 
 impl NodeViewContainer {
     fn new(octree: Arc<octree::Octree>) -> Self {
+        // Expensive I/O is performed by an I/O thread. Communication is realized through channels.
         let (node_id_sender, node_id_receiver) = mpsc::channel();
         let (node_data_sender, node_data_receiver) = mpsc::channel();
-        std::thread::spawn(move||{ 
+        std::thread::spawn(move||{
+            // Loads the next node data in the receiver queue.
             for node_id in node_id_receiver.into_iter() {
+                // We always request nodes at full resolution (i.e. not subsampled by the backend), because
+                // we can just as effectively subsample the number of points we draw in the client.
                 const ALL_POINTS_LOD: i32 = 1;                
                 let node_data = octree.get_node_data(&node_id, ALL_POINTS_LOD).unwrap();
+                // TODO: reshuffle?
```

Comment:
missing user name, feel free to add mine. Generally, the username should be of the author, not the asignee.